src/main/java/io/cdap/plugin/ariba/source/AribaStructuredTransformer.java

```diff
@@ -0,0 +1,366 @@
+        });
+        return jsonList.stream().map(
+          field -> readFields(field.get(recordSchema.getNonNullable().getComponentSchema().getRecordName()),
+                              recordSchema.getNonNullable().getComponentSchema())).collect(Collectors.toList());
+      } catch (JsonProcessingException e) {
+        LOG.error("Error in Processing Structured Transformer from Json, cause: {}", e.getMessage());
+      }
+    }
+    return Collections.emptyList();
+  }
+
+
+  private void processSchemaTypeValue(Schema fieldSchema, StructuredRecord.Builder recordBuilder,
+                                      String fieldName, Object fieldValue) {
+
+    // Get Non-nullable schema object for the current field
+    Schema nonNullSchema = fieldSchema.isNullable() ? fieldSchema.getNonNullable() : fieldSchema;
+    if (fieldValue instanceof StructuredRecord || fieldValue instanceof List) {
+      recordBuilder.set(fieldName, fieldValue);
+    } else {
+      boolean isBlank = handleBlankVal(recordBuilder, fieldName, (String) fieldValue,
+                                       nonNullSchema.getType() == Schema.Type.STRING);
+
+      // If value is NOT null or empty then transform acc. to the field's data type
+      if (!isBlank) {
+        try {
+          processValue(nonNullSchema, recordBuilder, fieldName, (String) fieldValue);
+        } catch (IOException e) {
+          LOG.error("Error in Processing Schema type value, cause: {}", e.getMessage());
+        }
+      }
+    }
+  }
+
+  @Nullable
+  @VisibleForTesting
+  Object getFieldNativeValue(JsonNode nativeRawRecord, String fieldName) {
+    JsonNode recordValue = nativeRawRecord.get(fieldName);
+    if (recordValue != null) {
+      if (nativeRawRecord.get(fieldName).getNodeType().name().equalsIgnoreCase(ResourceConstants.ARRAY) ||
+        nativeRawRecord.get(fieldName).getNodeType().name().equalsIgnoreCase(ResourceConstants.OBJECT)) {
+        return nativeRawRecord.get(fieldName);
+
+      } else if (nativeRawRecord.get(fieldName).getNodeType().name().equals(ResourceConstants.STRING) ||
+        nativeRawRecord.get(fieldName).getNodeType().name().equals(ResourceConstants.NUMBER) ||
+        nativeRawRecord.get(fieldName).getNodeType().name().equals(ResourceConstants.BOOLEAN)) {
+
+        return nativeRawRecord.get(fieldName).asText();
+
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Checks and sets null or original (whitespace filled) string as value in the
+   * {@code StructuredRecord.Builder} for field represented by {@code fieldName}
+   *
+   * @param recBuilder
+   * @param fieldName
+   * @param fieldValStr
+   * @param isTypeString
+   * @return True, if native value is null or empty. False, otherwise.
+   */
+  @VisibleForTesting
+  boolean handleBlankVal(StructuredRecord.Builder recBuilder, String fieldName,
+                                   @Nullable String fieldValStr, boolean isTypeString) {
```

Comment:
if the type of fieldValueStr is a string, why do you have a boolean param isTypeString?

Reply:
Fixed.
Removed it as it was not required, it was just a double check.